src/attributes/attr.js

```diff
@@ -97,7 +97,10 @@ if ( isIE ) {
 	};
 }
 
-jQuery.each( jQuery.expr.match.bool.source.match( /\w+/g ), function( _i, name ) {
+jQuery.each( (
+	"checked selected async autofocus autoplay controls defer disabled " +
+	"hidden ismap loop multiple open readonly required scoped"
+).split( " " ), function( _i, name ) {
```

Comment:
```suggestion
// HTML boolean attributes have special behavior:
// we consider the lowercase name to be the only valid value, so
// getting (if the attribute is present) normalizes to that, as does
// setting to any non-`false` value (and setting to `false` removes the attribute).
// See https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes
jQuery.each( (
	"checked selected async autofocus autoplay controls defer disabled " +
	"hidden ismap loop multiple open readonly required scoped"
).split( " " ), function( _i, name ) {
```

Note also that our list is outdated; it is missing several boolean attributes from [HTML Attributes](https://html.spec.whatwg.org/multipage/indices.html#attributes-3), includes a `scoped` that isn't even present there, and treats [`hidden`](https://html.spec.whatwg.org/multipage/interaction.html#the-hidden-attribute) as boolean despite the expansion of that attribute to be an enumeration including "until-found" by https://github.com/whatwg/html/commit/f5def65bda4d01a6a1e105fcf13fe45201b4f48a .

Reply:
Oh, interesting. Perhaps we should update this list for 4.0 as this can be considered a breaking change.

Or we should get rid of this manual normalization, depending on users doing the right thing here. The biggest issue are possibly falsy checks on values since an empty string is also a valid value. If we're worried about compat for such things, my guess would be we could limit this list to just a few entries like `checked` & `selected`. Thoughts, @timmywil @gibson042?

Reply:
I extracted the issue about the list itself into https://github.com/jquery/jquery/issues/5388; let's limit this PR to just the issue at hand and keep the list the same here.